Zend/zend_operators.c

```diff
@@ -1882,7 +1882,7 @@ ZEND_API int ZEND_FASTCALL concat_function(zval *result, zval *op1, zval *op2) /
 		size_t result_len = op1_len + op2_len;
 		zend_string *result_str;
 
-		if (UNEXPECTED(op1_len > ZSTR_MAX_LEN - op2_len)) {
+		if (UNEXPECTED(op1_len > ZSTR_MAX_LEN - op2_len || op2_len > ZSTR_MAX_LEN)) {
```

Comment:
How can op2_len > ZSTR_MAX_LEN? It's already coming from a zend_string, so that means there is a zend_string longer than ZSTR_MAX_LEN?

Reply:
Indeed. Thanks!